### PR TITLE
User Profile - rename search to suggest profile

### PR DESCRIPTION
**Pull request by ywangd**
This PR renames "search profile" to "suggest profile" in all relevant
places:

* Class names
* Variable names
* Docs
* URL and REST specs

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java [hunk 1]
```diff
@@ -191,7 +192,7 @@ public void searchProfile(SearchProfilesRequest request, ActionListener<SearchPr
                         "user_profile.user.full_name._2gram",
                         "user_profile.user.full_name._3gram",
                         "user_profile.user.email"
-                    ).type(MultiMatchQueryBuilder.Type.BOOL_PREFIX)
+                    ).type(MultiMatchQueryBuilder.Type.BOOL_PREFIX).fuzziness(Fuzziness.AUTO)
```
Comment (ywangd) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java:195]: The addition of `fuzziness` is not really about renaming. But it is too trivial to be on its own PR.

Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java]: On a second thought, I think we should leave it in a future PR after discussing whether we do want it. So I dropped it from this PR.

**Comment by ywangd** [comment]
Failure is unrelated but reproducible, I raised https://github.com/elastic/elasticsearch/issues/85198

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1-fips

**Review by sethmlarson — APPROVED** [review]
LGTM! :shipit:

**Comment by ywangd** [comment]
Failure is irrelevant, I reopenned https://github.com/elastic/elasticsearch/issues/85158

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-1

**Review by albertzaharovits — APPROVED** [review]
LGTM

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@elasticmachine update branch
